orion/controllers/programs/programsController.go

```diff
@@ -81,6 +84,42 @@ func DeleteProgram(c *gin.Context) {
   return
 }
 
-func CheckValidProgram() bool {
-  return true
+func CheckValidProgram(program Program) error {
+  // Retrieves the inputted values
+  programId := program.ProgramId
+  name := program.Name
+  grade1 := program.Grade1
+  grade2 := program.Grade2
+
+  // Checks if program name or program ID is empty
+  if programId == "" {
+    return errors.New("empty Program ID")
+  }
+
+  // TODO: use regex
+  // Checks if the program ID is alphanumeric
+  for _, i := range name {
+    if (i < 'a' || i > 'z') && (i < 'A' || i > 'Z') && (i < '1' || i > '0') && i != '_' {
+      return errors.New("invalid program name")
+    }
+  }
+
+  if name == "" {
+    return errors.New("empty program name")
+  }
+
+  // TODO: Check if program name is alphanumeric and can include spaces, underscores, and ampersands
+  // Checks if the program name is alphanumeric
+  for _, i := range name {
+    if (i < 'a' || i > 'z') && (i < 'A' || i > 'Z') && (i < '1' || i > '0') && i != '_' {
```

Comment:
you can delete this for loop and let max handle it. One is to avoid merge conflicts later. And two is because program name will have less restrictions.